transform/uuid_test.go

```diff
@@ -0,0 +1,118 @@
+package transform
+
+import (
+	"testing"
+
+	"github.com/buger/jsonparser"
+	uuid "github.com/satori/go.uuid"
+)
+
+func TestUuidWithFieldPresent(t *testing.T) {
+	spec := `{"sources": [{"path": "a.id"}, {"path": "a.type"}], "targetPath": "a.uuid", "delim": "-" }`
+	jsonIn := `{"a":{"author":"jason","id":2323223,"type":"tweet"}}`
+	jsonOut := `{"a":{"author":"jason","id":2323223,"type":"tweet","uuid":"2323223-tweet"}}`
+
+	cfg := getConfig(spec, false)
+	kazaamOut, _ := getTransformTestWrapper(UUID, cfg, jsonIn)
+
+	if kazaamOut != jsonOut {
+		t.Error("Transformed data does not match expectation.")
+		t.Log("Expected:   ", jsonOut)
+		t.Log("Actual:     ", kazaamOut)
+		t.FailNow()
+	}
+}
+func TestUUIDV4(t *testing.T) {
+	spec := `{"sources": [{"path": "a.id"}, {"path": "a.type"}], "targetPath": "a.uuid", "delim": "-",
+	"uuid": {"version": 4}}`
+	jsonIn := `{"a":{"author":"jason","id":2323223}`
+
+	cfg := getConfig(spec, false)
+	kazaamOut, _ := getTransformTestWrapper(UUID, cfg, jsonIn)
+	out, _, _, _ := jsonparser.Get([]byte(kazaamOut), "a", "uuid")
```

Comment:
I've done for UUIDv3 and UUIDv5 since I can know those exact UUID's for 4 i won't know the output.